examples/cloud-support/java_update_cases_blog_code/my-app/src/main/java/com/mycompany/app/App.java

```diff
@@ -0,0 +1,153 @@
+/**
+ * Copyright (C) 2022 Google Inc.
+ *
+ * <p>Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * <p>http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * <p>Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.mycompany.app;
+
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.http.HttpRequestInitializer;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.gson.GsonFactory;
+import com.google.api.services.cloudsupport.v2beta.CloudSupport;
+import com.google.api.services.cloudsupport.v2beta.model.CloudSupportCase;
+import com.google.api.services.cloudsupport.v2beta.model.ListCasesResponse;
+import com.google.auth.http.HttpCredentialsAdapter;
+import com.google.auth.oauth2.GoogleCredentials;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import javax.swing.JOptionPane;
+
+// starter code for cloud support API to initialize it
+public class App {
+
+  // Shared constants
+  static final String CLOUD_SUPPORT_SCOPE = "https://www.googleapis.com/auth/cloudsupport";
+
+  static String projectResource = "projects/";
+
+  public static void main(String[] args) {
+
+    try {
+      CloudSupport cloudSupport = getCloudSupportService();
+      // with the CloudSupport object, you may call other methods of the Support API
+      // for example, cloudSupport.cases().get("name of case").execute();
+
+      System.out.println("CloudSupport API is ready to use: " + cloudSupport);
+
+      projectResource =
+          projectResource + JOptionPane.showInputDialog("Please enter your project number: ");
```

Comment:
Opening a dialog forces the user to have a GUI to run your application. A better option is to pass a CLI argument